src/tools/rbd_wnbd/rbd_wnbd.cc

```diff
@@ -1670,8 +1232,10 @@ static int parse_args(std::vector<const char*>& args,
       cfg->remap_failure_fatal = true;
     } else if (ceph_argparse_flag(args, i, "--adapter-monitoring-enabled", (char *)NULL)) {
       cfg->adapter_monitoring_enabled = true;
-    } else if (ceph_argparse_witharg(args, i, &cfg->parent_pipe, err,
+    } else if (ceph_argparse_witharg(args, i, &parent_pipe, err,
                                      "--pipe-name", (char *)NULL)) {
+      std::cerr << "WARNING: '--pipe-name' has been deprecated and is currently ignored."
```

Comment:
This would make more sense _after_ the check on `err` IMO.  Otherwise, if there is an argparse error, it would fail on the option after having said that it's ignored.